chapter12/dags/nyc_dag.py

```diff
@@ -14,212 +14,198 @@
 from nyctransport.operators.s3_to_postgres import MinioPandasToPostgres
 from requests.auth import HTTPBasicAuth
 
-dag = DAG(
+with DAG(
     dag_id="nyc_dag",
-    schedule_interval="*/15 * * * *",
+    schedule="*/15 * * * *",
     start_date=pendulum.today("UTC").add(days=-1),
     catchup=False,
-)
+):
 
+    def _download_citi_bike_data(ts_nodash, **_):
+        citibike_conn = BaseHook.get_connection(conn_id="citibike")
 
-def _download_citi_bike_data(ts_nodash, **_):
-    citibike_conn = BaseHook.get_connection(conn_id="citibike")
+        url = f"http://{citibike_conn.host}:{citibike_conn.port}/recent/minute/15"
+        response = requests.get(url, auth=HTTPBasicAuth(citibike_conn.login, citibike_conn.password))
+        data = response.json()
 
-    url = f"http://{citibike_conn.host}:{citibike_conn.port}/recent/minute/15"
-    response = requests.get(url, auth=HTTPBasicAuth(citibike_conn.login, citibike_conn.password))
-    data = response.json()
+        s3_hook = S3Hook(aws_conn_id="s3")
+        s3_hook.load_string(
+            string_data=json.dumps(data),
+            key=f"raw/citibike/{ts_nodash}.json",
```

Comment:
all  instances with {{ ts_nodash }}
That should be changed to {{data_interval_start | ts_nodash}}